### ISSUE TITLE
Display test name in warning message when Monkeypatch.setenv/delenv use bytes keys

### ISSUE THREAD

**Issue by nicoddemus** [pytest-dev/pytest#4082]
From https://github.com/pytest-dev/pytest/pull/4056#issuecomment-427307239 by @gaborbernat: 

---

Hey guys, would it be possible to also report at least the test that catches this? In tox now I get:

```bash
018-10-05T09:23:49.0704702Z =============================== warnings summary ===============================
2018-10-05T09:23:49.0705517Z /home/vsts/work/1/s/.tox/py27/lib/python2.7/site-packages/_pytest/monkeypatch.py:232: PytestWarning: Environment variable value local('/tmp/pytest-of-vsts/pytest-0/popen-gw0/test_substitution_jenkins_cont0') should be str, converted to str implicitly
2018-10-05T09:23:49.0705945Z   value
2018-10-05T09:23:49.0706652Z /home/vsts/work/1/s/.tox/py27/lib/python2.7/site-packages/_pytest/monkeypatch.py:232: PytestWarning: Environment variable value 1 should be str, converted to str implicitly
```

I have 450+ tests though leaving me puzzled where to fix it...

---

Seems easy, `Monkeypatch` might receive an optional `nodeid` object that is used in the message.

**Comment by marcelotrevisani** [comment]
I think this was already fixed right?

**Comment by nicoddemus** [comment]
Note: We should also use the correct stack level

**Comment by nicoddemus** [comment]
Fixed by 5404246e64aa61f0f61e4825fe1cd3de963d4262
